diofant/integrals/tests/test_meijerint.py

```diff
@@ -651,3 +651,8 @@ def test_issue_6860():
 def test_issue_8368():
     assert meijerint_indefinite(cosh(x)*exp(-x*t), x) == (
         (-t - 1)*exp(x) + (-t + 1)*exp(-x))*exp(-t*x)/2/(t**2 - 1)
+
+
+def test_issue_4311():
+    assert meijerint_indefinite(x*abs(9 - x**2), x) == \
+        -81*meijerg(((3, 1), (2,)), ((1,), (2, 0)), x**2/9)
```

Comment:
Trivial evaluation with Piecewise:
```
In [1]: x = Symbol('x', real=True)

In [2]: integrate((x*abs(9 - x**2)).rewrite(Piecewise), x)
Out[2]: 
⎧  4      2                 
⎪ x    9⋅x         2        
⎪ ── - ────   for x  - 9 ≥ 0
⎪ 4     2                   
⎨                           
⎪   4      2                
⎪  x    9⋅x                 
⎪- ── + ────    otherwise   
⎩  4     2                  

In [3]: integrate(x*abs(9 - x**2), x)
Out[3]: 
            ⎛           │  2⎞
    ╭─╮1, 2 ⎜3, 1   2   │ x ⎟
-81⋅│╶┐     ⎜           │ ──⎟
    ╰─╯3, 3 ⎝ 1    2, 0 │ 9 ⎠

In [4]: _2.subs(x, 0).n()
Out[4]: 0

In [5]: _3.subs(x, 0).n()  # so, antiderivatives should be same
Out[5]: 0

In [6]: _2.subs(x, 1).n()
Out[6]: 4.25000000000000

In [7]: _3.subs(x, 1).n()  # but is seems they are not same
Out[7]: -3.16666666666667
```
and
```
In [10]: _3.subs(x, 4).n()
Out[10]: -66.1666666666667

In [11]: _2.subs(x, 4).n()
Out[11]: -8.00000000000000
```

@jksuom, do you care that the answer, you introduced with https://github.com/sympy/sympy/pull/11283 in SymPy is wrong?  That's from SymPy master:
```
$ ./bin/isympy -q


In [1]: integrate(x*abs(9 - x**2), x)
Out[1]: 
            ⎛           │  2⎞
    ╭─╮1, 2 ⎜3, 1   2   │ x ⎟
-81⋅│╶┐     ⎜           │ ──⎟
    ╰─╯3, 3 ⎝ 1    2, 0 │ 9 ⎠
```